pydantic/_internal/_generics.py

```diff
@@ -75,7 +75,7 @@ def __class_getitem__(cls, *args: Any) -> Any:
 
 if TYPE_CHECKING:
 
-    class DeepChainMap(ChainMap[KT, VT]):
+    class DeepChainMap(ChainMap[KT, VT]):  # type: ignore
```

Comment:
I'm not sure why Pyright complains about this since it's inside of a TYPE_CHECKING block, but the fix is easy enough...